cmake/CopyBuildProducts.cmake

```diff
@@ -0,0 +1,97 @@
+function(copy_build_products target)
+  set(valid_options "DEPENDENCY;DEPENDENCY_FILES;TARGET_LOCATION;CONDITION")
+  foreach(arg IN LISTS ARGN)
+    if("${arg}" IN_LIST valid_options)
+      set(mode "${arg}")
+    elseif("${mode}" STREQUAL "DEPENDENCY")
+      list(LENGTH COPY_BUILD_PRODUCTS_DEPENDENCIES current_dependency)
+      list(APPEND COPY_BUILD_PRODUCTS_DEPENDENCIES "${arg}")
+    elseif("${mode}" STREQUAL "DEPENDENCY_FILES")
+      list(APPEND "COPY_BUILD_PRODUCTS_${current_dependency}_FILES" "${arg}")
+    elseif("${mode}" STREQUAL "TARGET_LOCATION")
+      if(DEFINED "COPY_BUILD_PRODUCTS_${current_dependency}_TARGET_LOCATION")
+        message(FATAL_ERROR "Target location for dependency ${current_dependency} already set!")
+      endif()
+      set("COPY_BUILD_PRODUCTS_${current_dependency}_TARGET_LOCATION" "${arg}")
+    elseif("${mode}" STREQUAL "CONDITION")
+      if(DEFINED "COPY_BUILD_PRODUCTS_${current_dependency}_CONDITION")
+       set("COPY_BUILD_PRODUCTS_${current_dependency}_CONDITION" "${COPY_BUILD_PRODUCTS_${current_dependency}_CONDITION} ${arg}")
+      else()
+        set("COPY_BUILD_PRODUCTS_${current_dependency}_CONDITION" "${arg}")
+      endif()
+    endif()
+  endforeach()
+
+  set(COPY_BUILD_PRODUCTS_COPIED_RESOURCES)
+  foreach(dependency_idx RANGE 0 ${current_dependency})
+    list(GET COPY_BUILD_PRODUCTS_DEPENDENCIES ${dependency_idx} dependency)
+    get_target_property(type ${dependency} TYPE)
+    get_target_property(resx_tmp ${dependency} RESOURCE)
+    get_target_property(bindir ${dependency} BINARY_DIR)
+
+    set(resx "")
+    if(resx_tmp)
+      foreach(res IN LISTS resx_tmp)
+        if(IS_ABSOLUTE "${res}")
+          file(RELATIVE_PATH relpath "${bindir}" "${res}")
+          list(APPEND resx "${relpath}")
+        else()
+          list(APPEND resx "${res}")
+        endif()
+      endforeach()
+    endif()
+
+    set(cond_succeeded NO)
+    if(NOT DEFINED "COPY_BUILD_PRODUCTS_${dependency_idx}_CONDITION")
+      set(cond_succeeded YES)
+    else()
+      cmake_language(EVAL CODE "if(COPY_BUILD_PRODUCTS_${dependency_idx}_CONDITION)\nset(cond_succeeded YES)\nendif()")
```

Comment:
I have no idea if this can be done in a better way. I tried multiple times and I kept getting the wrong result.

Chances are, it's probably still wrong but it Worked On My Machine:tm:.